torch/csrc/autograd/python_variable.cpp

```diff
@@ -227,7 +227,7 @@ int THPVariable_set_data(THPVariable *self, PyObject *data)
     throw torch::TypeError("Variable data has to be a tensor, but got %s", Py_TYPE(data)->tp_name);
   }
 
-  self->cdata.set_data(THPVariable_UnpackData(data));
+  self->cdata.set_data(THPVariable_Unpack(data));
```

Comment:
It's okay to use `THPVariable_Unpack()` instead of  `THPVariable_UnpackData()` here, because `set_data(tensor)` ignores `tensor`'s AutogradMeta.